src/ConnectionFactory.php

```diff
@@ -83,4 +95,31 @@ private function registerTypes(ContainerInterface $container): void
             Type::addType($name, $className);
         }
     }
+
+    /**
+     * @param array<array-key, mixed> $params
+     *
+     * @return array<array-key, mixed>
+     */
+    private function parseDatabaseUrl(
+        #[SensitiveParameter]
+        array $params,
+    ): array {
+        if (! isset($params['url'])) {
+            return $params;
+        }
+
+        $parser = new DsnParser();
+
+        $parsedParams = $parser->parse($params['url']);
+        unset($params['url']);
+
+        $result = array_merge($params, $parsedParams);
+
+        if (isset($result['driver']) && isset($result['driverClass'])) {
+            unset($result['driverClass']);
```

Comment:
This also needs a test

Reply:
This line is covered by the test because `driverClass` is always defined by default.

https://github.com/Roave/psr-container-doctrine/blob/0b41fd773f45e305aba1b8d02a3361194e9b04d2/src/ConnectionFactory.php#L55

On the other hand, we can remove this statement since the `driver` param has a higher priority. But this behaviour can be changed in future versions of `doctrine/dbal`.